flink-connector-cassandra/src/main/java/org/apache/flink/connector/cassandra/source/split/CassandraSplit.java

```diff
@@ -48,6 +52,30 @@ public CassandraSplitState toSplitState() {
         return new CassandraSplitState(new HashSet<>(ringRanges), splitId());
     }
 
+    public void serialize(ObjectOutputStream objectOutputStream) throws IOException {
```

Comment:
let's make this package private. Personally I'd move all of it into the serializer, but I can see why it could be useful to have it here.
For symmetry I'd prefer this method to be static and accept a CassandraSplit instance.

Reply:
this method can be package private only if it resides in the serializer and it makes sense that it takes a CassandraSplit instance only if it resides in the serializer. So I'll move the method and make it static

Reply:
same for deserialize and EnumeratorStateSerializer

Reply:
> For symmetry I'd prefer this method to be static and accept a CassandraSplit instance 

I made the serializer take an OutputStream to avoid creating an OutputStream for each split in the EnumeratorState serialization which would create a stream header for each split